frontends/concrete-python/concrete/fhe/extensions/multivariate.py

```diff
@@ -0,0 +1,104 @@
+"""
+Declaration of `multivariate` extension.
+"""
+
+from copy import deepcopy
+from typing import Any, Callable, Optional, Type, Union
+
+import numpy as np
+
+from ..dtypes import BaseDataType, Float
+from ..representation import Node
+from ..tracing import ScalarAnnotation, Tracer
+from ..values import ValueDescription
+
+
+def multivariate(
+    function: Callable,
+    outputs: Optional[Union[BaseDataType, Type[ScalarAnnotation]]] = None,
+) -> Callable:
+    """
+    Wrap a multivariate function so that it is traced into a single generic node.
+
+    Args:
+        function (Callable[[Any, ...], Any]):
+            multivariate function to wrap
+
+        outputs (Optional[Union[BaseDataType, Type[ScalarAnnotation]]], default = None):
+            data type of the result, unused during compilation, required for direct definition
+
+    Returns:
+        Callable[[Union[Tracer, Any], ...], Union[Tracer, Any]]:
+            another multivariate function that can be called with Tracers as well
+    """
+
+    def wrapper(*args: Union[Tracer, Any]) -> Union[Tracer, Any]:
+        """
+        Evaluate or trace wrapped multivariate function.
+
+        Args:
+            args (Tuple[Union[Tracer, Any]]):
+                inputs of the function
+
+        Returns:
+            Union[Tracer, Any]:
+                result of tracing or evaluation
+        """
+
+        if any(isinstance(arg, Tracer) for arg in args):
```

Comment:
I would revert the test to have one indentation level removed from most of the code.
``python
any_tracer = any(isinstance(arg, Tracer) for arg in args)
if not any_tracer:
    return function(...)
# real code now less indented
``

Reply:
Done.